### PR TITLE
feat: Restful support for BM25 function

### PR DESCRIPTION
**Pull request by zhengbuqian**
issue: https://github.com/milvus-io/milvus/issues/35853

### CODE REVIEW

File: internal/distributed/proxy/httpserver/handler_v2.go [hunk 1]
```diff
@@ -1159,8 +1185,39 @@ func (h *HandlersV2) createCollection(ctx context.Context, c *gin.Context, anyRe
...
+
+		for _, function := range httpReq.Schema.Functions {
+			functionTypeValue, ok := schemapb.FunctionType_value[function.FunctionType]
+			if !ok {
+				log.Ctx(ctx).Warn("function's data type is invalid(case sensitive).", zap.Any("function.DataType", function.FunctionType), zap.Any("function", function))
+				HTTPAbortReturn(c, http.StatusOK, gin.H{
+					HTTPReturnCode:    merr.Code(merr.ErrParameterInvalid),
+					HTTPReturnMessage: merr.ErrParameterInvalid.Error() + ", function data type " + function.FunctionType + " is invalid(case sensitive).",
+				})
+				return nil, merr.ErrParameterInvalid
```
Comment (smellthemoon) [inline internal/distributed/proxy/httpserver/handler_v2.go:1202]: It is recommended not to throw err nakedly, but to use wrapErrXXX to carry necessary information.

File: internal/distributed/proxy/httpserver/constant.go [hunk 2]
```diff
@@ -94,14 +94,22 @@ const (
...
-	HTTPReturnDescription       = "description"
+	HTTPReturnFieldName             = "name"
+	HTTPReturnFieldID               = "id"
+	HTTPReturnFieldType             = "type"
+	HTTPReturnFieldPrimaryKey       = "primaryKey"
+	HTTPReturnFieldPartitionKey     = "partitionKey"
+	HTTPReturnFieldAutoID           = "autoId"
+	HTTPReturnFieldElementType      = "elementType"
+	HTTPReturnDescription           = "description"
+	HTTPReturnFieldIsFunctionOutput = "is_function_output"
```
Comment (smellthemoon) [inline internal/distributed/proxy/httpserver/constant.go:105]: It seems more appropriate to use isFunctionOutput to keep it consistent with other forms.

File: internal/distributed/proxy/httpserver/handler_v2.go [hunk 3]
```diff
@@ -1083,6 +1098,17 @@ func (h *HandlersV2) createCollection(ctx context.Context, c *gin.Context, anyRe
...
 	if httpReq.Schema.Fields == nil || len(httpReq.Schema.Fields) == 0 {
+		if len(httpReq.Schema.Functions) > 0 {
+			err := merr.WrapErrParameterInvalid("schema", "functions",
+				"functions are not supported for quickly create collection")
+			log.Ctx(ctx).Warn("high level restful api, quickly create collection fail", zap.Error(err), zap.Any("request", anyReq))
+			HTTPAbortReturn(c, http.StatusOK, gin.H{
+				HTTPReturnCode:    merr.Code(err),
+				HTTPReturnMessage: err.Error(),
+			})
+			return nil, err
```
Comment (smellthemoon) [inline internal/distributed/proxy/httpserver/handler_v2.go:1109]: L1102-L1109 is no ut coverage. If it is convenient, could you add  some ut?

File: internal/distributed/proxy/httpserver/utils.go [hunk 4]
```diff
@@ -685,8 +723,8 @@ func anyToColumns(rows []map[string]interface{}, validDataMap map[string][]bool,
 			IsDynamic: field.IsDynamic,
 		}
 	}
-	if len(nameDims) == 0 {
-		return nil, fmt.Errorf("collection: %s has no vector field", sch.Name)
+	if len(nameDims) == 0 && len(sch.Functions) == 0 {
+		return nil, fmt.Errorf("collection: %s has no vector field or functions", sch.Name)
```
Comment (czs007) [inline internal/distributed/proxy/httpserver/utils.go:727]: or ==> and

**Pull request merged**